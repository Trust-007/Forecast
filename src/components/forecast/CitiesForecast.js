import { useSelector } from "react-redux";
import CityForecast from "./CityForecast";
import classes from "./CitiesForecast.module.css";

const CitiesForecast = () => {
  const forecast = useSelector((state) => state.forecast.list);
  const search = useSelector((state) => state.forecast.search);

  let hi = forecast;
  if (search !== "") {
    hi = forecast.filter((item) => {
      if (item.Country.LocalizedName.toLowerCase() === search.toLowerCase()) {
        return item;
      }
      return "";
    });
  }
  const list = hi.map((item) => {
    return (
      <CityForecast
        key={item.Key}
        id={item.Key}
        city={item.AdministrativeArea.LocalizedName}
        lat={item.GeoPosition.Latitude}
        long={item.GeoPosition.Longitude}
      />
    );
  });
  let content;
  if (list.length === 0) {
    content = <p>No items</p>;
  } else {
    content = list;
  }
  return <div className={classes.container}>{content}</div>;
};

export default CitiesForecast;
