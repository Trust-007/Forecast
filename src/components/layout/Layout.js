import NavHeader from './NavHeader';
import classes from './Layout.module.css';

const Layout = (props) => {
  const { children } = props;
  return (
    <div>
      <NavHeader />
      <div className={classes.container}>{children}</div>
    </div>
  );
};

export default Layout;
