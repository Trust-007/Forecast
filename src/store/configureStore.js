import { configureStore } from '@reduxjs/toolkit';
import forecastSlice from './forecast/forecastSlice';

const store = configureStore({
  reducer: {
    forecast: forecastSlice,
  },
});

export default store;
