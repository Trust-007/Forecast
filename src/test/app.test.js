import React from 'react';

import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../store/configureStore';
import getData from '../store/forecast/actionThunk';

import CitiesForecast from '../components/forecast/CitiesForecast';
import Layout from '../components/layout/Layout';

describe('display the citie in the page', () => {
  it('should display all the cities', () => {
    const { container } = render(
      <Provider store={store}>
        <CitiesForecast />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});

describe('should show the navigation', () => {
  it('full app rendering/navigating', async () => {
    render(
      <Provider store={store}>
        <Layout />
      </Provider>,
      { wrapper: BrowserRouter },
    );
    expect(screen.queryByText(/Canada/)).toMatchSnapshot();
  });
});

describe('Forecast reducers', () => {
  jest.setTimeout(30000);
  it('should fetch items ', async () => {
    await store.dispatch(getData());
    expect(store.getState().forecast.list.length).toEqual(14);
  });
  it('should be an empty string', () => {
    expect(store.getState().forecast.search).toBe('');
  });
});
