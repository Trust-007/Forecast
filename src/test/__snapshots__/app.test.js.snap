// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display the citie in the page should display all the cities 1`] = `
<div>
  <div
    class="container"
  >
    <p>
      No items
    </p>
  </div>
</div>
`;

exports[`display the rockets in the page should display all the rockets 1`] = `
<div>
  <div
    class="container"
  >
    <p>
      No items
    </p>
  </div>
</div>
`;

exports[`should show the navigation full app rendering/navigating 1`] = `null`;
